torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -55,6 +58,29 @@ struct TORCH_API WorkerInfo {
   const worker_id_t id_;
 };
 
+// Struct that stores all the metadata needed to retry a given RPC.
+struct TORCH_API RpcRetryInfo {
+  RpcRetryInfo(
+      const WorkerInfo& to,
+      Message&& message,
```

Comment:
Does this mean we actually need multiple copies of the message? Every time we retry send, we need to pass a rvalue message to the `send` API. Is there a way to just use one copy of the message?

Reply:
btw, I think any send-with-retry api in general is going to require at least 1 copy per try (including the original try).

Tensor copies should generally be cheap due to refcounting (maybe I'm missing a case?).
Payload will have to be copied, though typically our Message sizes are dominated by tensors, so might not be as bad as it sounds.

It's good that you called out the double std::move() calls here, though - those would have been unfortunate (and probably mean that we probably could have better test coverage of this new api).